### DISCUSSION TITLE
[Idea] Centralized Folder for Dynamic Files in Horizontal Scaling Environments

### DISCUSSION THREAD

**Discussion by pvujic** [PrestaShop/PrestaShop#36950] [Ideas]
### The problem

In horizontal scaling environments, PrestaShop users often need to perform significant prep work to ensure that dynamically changing files—such as CSVs from modules, logs, mails, and translations—are accessible across multiple servers. These files are typically scattered across different directories, which makes it difficult to ensure compatibility when using shared storage solutions like NFS.

The current directory structure looks something like this:

/mails
/downloads
/img
/sitemaps
/themes/theme-name/mails
/modules/module-name/csv
and many more...

This structure results in a fragmented organization of dynamic files, creating challenges in multiserver setups that use network file systems (NFS).

### The solution

The proposed solution is to introduce a centralized folder in the PrestaShop platform, specifically for dynamic files. This folder would be accessible to module creators and would be used for uploading CSVs, logs, and other frequently changing files. By centralizing these files, we can ensure better compatibility and ease of management in horizontal scaling environments using shared file systems like NFS.

Changes would include:

Creation of a /shared directory in PrestaShop’s root or htdocs folder to act as the central hub for dynamic files.
Subdirectories such as /shared/modules, /shared/mails, /shared/translations where module-specific and general dynamic data can be stored.

This would simplify maintenance and improve compatibility for users with multiserver configurations.

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Hlavtox** [comment]
Check this https://github.com/PrestaShop/ADR/pull/34

@kpodemski I think we still have time to do this for v9, it's just matter of defining the path and saying where to store this.
